src/types.ts

```diff
@@ -80,367 +52,10 @@ export type ProximityRule = {
-  reverseGeocodingTypes?: TypeRule[];
-
-  /**
-   * Automatically selects the first feature in the result list.
-   *
-   * Default: `true`.
-   */
-  selectFirst?: boolean;
-
-  /**
-   * Indicates when to show place/POI types in the dropdown:
-   * - `"never"`: Hide the type.
-   * - `"always"`: Always show the type.
-   * - `"if-needed"`: Show the type only if it cannot be determined from the icon.
-   *
-   * Default: `"if-needed"`.
-   */
-  showPlaceType?: ShowPlaceType;
-
-  /**
-   * Displays results while typing:
-   * - `false`: Search occurs only on pressing the Enter key.
-   * - `true`: Search begins when the input meets the `minLength` requirement.
-   *
-   * Default: `true`.
-   */
-  showResultsWhileTyping?: boolean;
-
-  /**
-   * Types to query, either as an array or `[minZoom, maxZoom, type]` format.
-   * `minZoom` is inclusive, `maxZoom` is exclusive, and either can be `null` or `undefined` for unbounded values.
-   *
-   * See `reverseGeocodingTypes` option for reverse geocoding types.
-   *
-   * Default: `undefined` (uses server default feature types).
-   */
-  types?: TypeRule[];
-
-  /**
-   * Specifies the zoom level to animate the map to for a geocoded result when no bounding box is present or when the result is a point.
-   * If a bounding box is present and not a point, the map will fit to the bounding box.
-   *
-   * Values are key-value pairs where the key is a `<type>` or `<type>.<category>` and the value is the zoom level.
-   *
-   * Default: `GeocodingControl.ZOOM_DEFAULTS`.
-   */
-  zoom?: Record<string, number>;
-};
-
-export type PickedResultStyle =
-  | "marker-only"
-  | "full-geometry"
-  | "full-geometry-including-polygon-center-marker";
+export type PickedResultStyle = "marker-only" | "full-geometry" | "full-geometry-including-polygon-center-marker";
 
 export type EnableReverse = "never" | "always" | "button";
 
 export type ShowPlaceType = "never" | "always" | "if-needed";
 
-export type DispatcherTypeCC = {
-  featuresListed: { features: Feature[] | undefined };
-  featuresMarked: { features: Feature[] | undefined };
-  optionsVisibilityChange: { optionsVisible: boolean };
-  pick: { feature: Feature | undefined };
-  queryChange: { query: string };
-  response: { url: string; featureCollection: FeatureCollection };
-  reverseToggle: { reverse: boolean };
-  select: { feature: Feature | undefined };
-};
-
-export type DispatcherType = {
-  [T in keyof DispatcherTypeCC as Lowercase<T>]: DispatcherTypeCC[T];
-};
-
-export type RedefineType<
-  OriginalType,
-  UpdatedType extends { [K in keyof OriginalType]: OriginalType[K] } & {
-    [K in Exclude<keyof UpdatedType, keyof OriginalType>]: never;
-  },
-> = UpdatedType;
-
-export type TypeRule =
-  | string
-  | [
-      minZoom: number | null | undefined,
-      maxZoom: number | null | undefined,
-      type: string,
-    ];
+export type TypeRule = string | [minZoom: number | null | undefined, maxZoom: number | null | undefined, type: string];
```

Comment:
Maybe this could be updated to include the actual types you can use? eg `"locality"`, `"poi"`?